rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1850,6 +1913,135 @@ public Response deleteStudentContactLogEntry(@PathParam("STUDENTID") Long studen
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+
+    if (!restSecurity.hasPermission(new String[] { StudentPermissions.FIND_STUDENT, UserPermissions.USER_OWNER }, student, Style.OR)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    
+    if (staffMember == null) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    ContactLogAccess access = studentController.resolveContactLogAccess(student);
+    
+    if (!access.equals(ContactLogAccess.ALL)) {
+      if (access.equals(ContactLogAccess.OWN)) {
+        if (contactLogEntry.getCreator() != null) {
+          if (!contactLogEntry.getCreator().getId().equals(staffMember.getId())) {
+            return Response.status(Status.FORBIDDEN).build();
+          }
+        }
+      } else {
+        return Response.status(Status.FORBIDDEN).build();
+      }
+    }
+    StudentContactLogEntryComment contactLogEntryComment = studentContactLogEntryCommentController.createContactLogEntryComment(contactLogEntry, entity.getText(), entity.getCommentDate(), loggedUser.getFirstName() + " " + loggedUser.getLastName(), staffMember);
+    return Response.ok(objectFactory.createModel(contactLogEntryComment)).build();
+  }
+  
+  @Path("/students/{STUDENTID:[0-9]*}/contactLogEntries/{ENTRYID:[0-9]*}/comments/{COMMENTID:[0-9]*}")
+  @PUT
+  @RESTPermit(handling = Handling.INLINE)
+  public Response updateStudentContactLogEntryComment(@PathParam("STUDENTID") Long studentId, @PathParam("ENTRYID") Long entryId, @PathParam("COMMENTID") Long commentId,
+      fi.otavanopisto.pyramus.rest.model.StudentContactLogEntryCommentRestModel entity) {
+    if (entity == null) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+    
+    StudentContactLogEntryComment contactLogEntryComment = studentContactLogEntryCommentController.findContactLogEntryCommentById(commentId);
+    if (contactLogEntryComment == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    StudentContactLogEntry contactLogEntry = contactLogEntryComment.getEntry();
+
+    if (contactLogEntry == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    if (contactLogEntry.getArchived()) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
```

Comment:
Am I blind or is this comparing `contactLogEntry.getStudent().getId()` to itself?